docs/architecture/adr-019-protobuf-state-encoding.md

```diff
@@ -299,6 +308,21 @@ func (msg MsgSubmitEvidence) GetEvidence() eviexported.Evidence {
 }
 ```
 
+### Amino Compatibility
+
+Our custom implementation of `Any` can be used transparently with Amino if used
+with the proper codec instance. What this means is that interfaces packed within
+`Any`s will be amino marshaled like regular Amino interfaces (assuming they
+have been registered properly with Amino).
+
+In order for this functionality to work:
+* **all legacy code must use `*codec.Codec` instead of `*amino.Codec` which is
+now a wrapper which properly handles `Any`**
+* **all new code should use `Marshaler` which is compatible with both amino and
+protobuf**
+
+Also, before v0.39, `codec.Codec` will be renamed to `codec.LegacyAmino`.
```

Comment:
we should this as a deprecation note on the `codec.Codec` godoc too.

Reply:
Adding a deprecated note to the godoc